test/main-tests/test.js

```diff
@@ -242,4 +242,22 @@ describe('EslintValidationFilter', function describeEslintValidationFilter() {
       }
     });
   });
+
+  it('throws when `throwOnError` is set and result severity', function(done) {
+    this.timeout(0);
+
+    const promise = shouldReportErrors(FIXTURES, {
+      options: {
+        ignore: false,
+      },
+      throwOnError: true
+    })();
+
+    return promise.catch((err) => {
+      expect(err).to.be.an('error');
+      expect(err.message).to.equal('rules violation with `error` severity level');
+    });
```

Comment:
Note that the assertions won't run when the promise does not throw. Maybe we should use chai-as-promised here.